### PR TITLE
#456 : Commands to Manage operations on free call users

### PR DESCRIPTION
**Pull request by anandrgitnirman**
View the full list of users who have made free calls to your service and also the usage ( i.e how many free calls)

### CODE REVIEW

File: snetd/cmd/free_call_users.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+// initializes and returns the new unlock user of free calls command object
+func newFreeCallUserUnLockCommandCommand(cmd *cobra.Command, args []string, pComponents *Components) (command Command, err error) {
+	userID, err := getUserId(cmd)
+	if err != nil {
+		return
+	}
+	command = &freeCallUserUnLockCommand{
+		lockStorage: pComponents.LockerStorage(),
+		userId:      userID,
+		component:   pComponents,
```
Comment (vsbogd) [inline snetd/cmd/free_call_users.go]: Why don't keep organization metadata instead of `pComponents`?

File: snetd/cmd/free_call_users.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+func newFreeCallResetCountCommand(cmd *cobra.Command, args []string, pComponents *Components) (command Command, err error) {
+	userID, err := getUserId(cmd)
+	if err != nil {
+		return
+	}
+	command = &freeCallUserResetCountCommand{
+		lockStorage: pComponents.LockerStorage(),
+		userStorage: pComponents.FreeCallUserStorage(),
+		userId:      userID,
+		component:   pComponents,
```
Comment (vsbogd) [inline snetd/cmd/free_call_users.go]: Why don't keep organization metadata instead of pComponents?

File: snetd/cmd/free_call_users.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+// initializes and returns the new unlock user of free calls command object
+func newFreeCallUserUnLockCommandCommand(cmd *cobra.Command, args []string, pComponents *Components) (command Command, err error) {
+	userID, err := getUserId(cmd)
+	if err != nil {
+		return
+	}
+	command = &freeCallUserUnLockCommand{
+		lockStorage: pComponents.LockerStorage(),
```
Comment (vsbogd) [inline snetd/cmd/free_call_users.go]: As far as I understand `FreeCallUserService` uses another lock storage: https://github.com/singnet/snet-daemon/blob/06bb92bb0ffa4811cbac50e1809f5f391657f62f/snetd/cmd/components.go#L258

File: snetd/cmd/free_call_users.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+func (command *freeCallUserUnLockCommand) unlockFreeCallUser() (err error) {
+	key := &escrow.FreeCallUserKey{}
+	key.UserId = freeCallUserId
+	key.OrganizationId = config.GetString(config.OrganizationId)
+	key.ServiceId = config.GetString(config.ServiceId)
+	key.GroupID = command.component.OrganizationMetaData().GetGroupIdString()
+	// check whether the key exists or not
+	_, ok, err := command.lockStorage.Get(key.String())
+	if !ok {
+		fmt.Printf("Error: Free Call user %s not found\n", key.String())
```
Comment (vsbogd) [inline snetd/cmd/free_call_users.go]:
```suggestion
		fmt.Printf("Error: Free Call lock for user %s is not found\n", key.String())
```

File: snetd/cmd/free_call_users.go [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+	err = command.lockStorage.Delete(key.String())
+	if err != nil {
+		fmt.Printf("Error: Unable to unlock the user -%s\n", key.String())
+		return
+	}
+	fmt.Printf("Success: User %s unlocked\n", key.String())
+	return
+}
+
+// release the lock on the user with the given user id
```
Comment (vsbogd) [inline snetd/cmd/free_call_users.go]:
```suggestion
// reset free locks counter for a given user id
```

File: snetd/cmd/free_call_users.go [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+// initializes and returns the new reset count for free call user command object
+func newFreeCallResetCountCommand(cmd *cobra.Command, args []string, pComponents *Components) (command Command, err error) {
+	userID, err := getUserId(cmd)
+	if err != nil {
+		return
+	}
+	command = &freeCallUserResetCountCommand{
+		lockStorage: pComponents.LockerStorage(),
```
Comment (vsbogd) [inline snetd/cmd/free_call_users.go]: Is not used as far as I see:
```suggestion
```

File: snetd/cmd/free_call_users.go (hunk 3 above)
Comment (vsbogd) [reply snetd/cmd/free_call_users.go]: I would suggest adding new component `FreeCallLockStorage` and use it here and in `components.FreeCallUserService` method

File: snetd/cmd/free_call_users.go (hunk 1 above)
Comment (anandrgitnirman) [reply snetd/cmd/free_call_users.go]: makes sense , we can get away from Components here

**Pull request merged**